src/pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -185,7 +185,7 @@ def __init__(
         self._layer_sync: Optional[LayerSync] = NativeSyncBatchNorm() if sync_batchnorm else None
         self.checkpoint_io: Optional[CheckpointIO] = None
         self._amp_type_flag: Optional[LightningEnum] = None
-        self._amp_level_flag: Optional[str] = amp_level
+        self._amp_level_flag: str = amp_level or "O2"
```

Comment:
I don't think this should be set here, because the precision plugin is not yet selected so we don't know if O2 is even applicable.

Reply:
updated!